gcc/rust/backend/rust-compile-base.cc

```diff
@@ -26,14 +26,21 @@
 
 #include "fold-const.h"
 #include "stringpool.h"
+#include "attribs.h"
 
 namespace Rust {
 namespace Compile {
 
+bool inline should_mangle_item (const tree fndecl)
+{
+  return lookup_attribute ("no_mangle", DECL_ATTRIBUTES (fndecl)) == NULL_TREE;
```

Comment:
I really like how you did this to split up the logic.

I guess the GCC idiom would be that this should be a macro called SHOULD_MANGLE_ITEM_P and put it over in rust-tree.h not sure what design I want in the code-generation part at the moment so lets stick with this for now.

Reply:
> I really like how you did this to split up the logic.
> 
> I guess the GCC idiom would be that this should be a macro called SHOULD_MANGLE_ITEM_P and put it over in rust-tree.h not sure what design I want in the code-generation part at the moment so lets stick with this for now.

Noted. I will keep this in mind in the near future.